### DISCUSSION TITLE
"no viable alternative at input ..." for simple struct creation

### DISCUSSION THREAD

**Discussion by pcholakov** [p-org/P#661] [Q&A] [answered]
Hey folks!

Amazing project, it's been on my list of things to play with for a long time but only finally getting to it now! Simple question that I couldn't figure out the answer for myself. I have the following code:

```
event PromiseInit: (id: int);

...

fun setup(...) {
  var i: int;
  ...
  announce PromiseInit, (id = i);
}
```

This ends up in the following compilation error:

```
[Parser Error:]
[TestDriver.p] parse error: line 24:35 no viable alternative at input 'id=i)'
```

I was able to work around the problem by creating a local variable of the required type as follows:

```
fun setup(...) {
  var i: int;
  var promiseId: (id: int);
  ...
  promiseId.id = i;
  announce PromiseInit, promiseId;
}
```

Is there a more elegant way to do this?

I am using P installed from Homebrew on macOS:

```
❯ p --version
P version 2.0.22.0
~~ [PTool]: Thanks for using P! ~~
```

Thanks in advance and very much looking forward to getting deeper with P!

**Comment by ankushdesai** [comment, marked as answer]
Please replace `announce PromiseInit, (id = i);` with `announce PromiseInit, (id = i,);`. For the compiler to distinguish single field tuple values from `()`. thanks!

Let us know your feedback as you explore P more. @pcholakov

**Reply by pcholakov** [reply]
Thank for the quick answer!
